src/cadenza.js

```diff
@@ -154,26 +154,29 @@ export class CadenzaClient {
    * @param {boolean} [options.debug]
    */
   constructor(baseUrl, { debug = false, iframe, webApplication } = {}) {
```

Comment:
The baseUrl is now also optional. Please update the docs accordingly.

Reply:
That would be nice, but it would be incompatible change, we would need to release a new major version. (and we want this in 9.4 as well)

Reply:
We can define overloads for the function and deprecate the one with the baseUrl param:
```
diff --git a/CHANGELOG.md b/CHANGELOG.md
index 68a9392..a834a98 100644
--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -6,8 +6,12 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## Unreleased
 ### Added
+- `cadenza()` overload with a single options parameter
 - `invalidateCaches` option for `CadenzaClient#reload`
 
+### Deprecated
+- `cadenza()` overload with the `baseUrl` and `options` parameters
+
 ## 2.7.0 - 2024-04-17
 
 ## 2.6.0 - 2024-04-12
diff --git a/src/cadenza.js b/src/cadenza.js
index 8c46d85..c220f6d 100644
--- a/src/cadenza.js
+++ b/src/cadenza.js
@@ -1,20 +1,44 @@
 /**
- * Creates an instance of the Cadenza JS client.
- *
- * @param {string} baseUrl - The base URL of the Cadenza server
- * @param {object} [options] - Options
- * @param {HTMLIFrameElement | string} [options.iframe] - An iframe for embedding Cadenza or the iframe's ID.
+ * @typedef CadenzaClientOptions
+ * @property {string} [baseUrl] - The base URL of the Cadenza server
+ * @property {HTMLIFrameElement | string} [options.iframe] - An iframe for embedding Cadenza or the iframe's ID.
  *   The iframe is required only for methods that embed Cadenza in an iframe, so e.g. not for {@link CadenzaClient#fetchData}.
  *   If you want to embed Cadenza in multiple iframes, you need to create an instance of the `CadenzaClient` per iframe.
- * @param {ExternalLinkKey} [options.webApplication] - An external link that Cadenza uses to resolve the
+ * @property {ExternalLinkKey} [options.webApplication] - An external link that Cadenza uses to resolve the
  *   [target origin](https://developer.mozilla.org/en-US/docs/Web/API/Window/postMessage#targetorigin) when posting events.
  *   This is required if Cadenza and your application are not running on the same origin.
  *   Please ensure that the user has view privilege for that link!
- * @param {boolean} [options.debug] - Whether to enable debug logging
+ * @property {boolean} [options.debug] - Whether to enable debug logging
+ */
+
+/**
+ * Creates an instance of the Cadenza JS client.
+ *
+ * @overload
+ * @param {CadenzaClientOptions} [options] - Options
+ * @return {CadenzaClient}
+ * @throws For invalid arguments
+ */
+/**
+ * Creates an instance of the Cadenza JS client.
+ *
+ * @deprecated This overload is deprecated since <time>2024-05-08</time> and will eventually be removed.
+ *   Please use the other overload.
+ * @overload
+ * @param {string} baseUrl - The base URL of the Cadenza server
+ * @param {Exclude<CadenzaClientOptions, 'baseUrl'>} [options] - Options
+ * @return {CadenzaClient}
  * @throws For invalid arguments
  */
-export function cadenza(baseUrl, options) {
-  return new CadenzaClient(baseUrl, options);
+/**
+ * @param {string | CadenzaClientOptions} [baseUrlOrOptions]
+ * @param {CadenzaClientOptions} [options]
+ */
+export function cadenza(baseUrlOrOptions, options) {
+  if (typeof baseUrlOrOptions === 'string') {
+    options = { baseUrl: baseUrlOrOptions, ...options };
+  }
+  return new CadenzaClient(options);
 }
 
 /* @ts-ignore */
@@ -147,13 +171,9 @@ export class CadenzaClient {
 
   /**
    * @hidden
-   * @param {string} baseUrl
-   * @param {object} [options]
-   * @param {HTMLIFrameElement | string} [options.iframe]
-   * @param {ExternalLinkKey} [options.webApplication]
-   * @param {boolean} [options.debug]
+   * @param {CadenzaClientOptions} [options]
    */
-  constructor(baseUrl, { debug = false, iframe, webApplication } = {}) {
+  constructor({ baseUrl = '', debug = false, iframe, webApplication } = {}) {
     assert(validUrl(baseUrl), `Invalid baseUrl: ${baseUrl}`);
     if (webApplication) {
       assert(

```

Reply:
Ah, very nice, yes, good idea @jkissel :+1: 
Can you make this code change @MrLynx93 ?

Reply:
I added it